src/DocumentFormat.OpenXml/SimpleTypes/OpenXmlComparableReferenceType.cs

```diff
@@ -0,0 +1,179 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+
+namespace DocumentFormat.OpenXml
+{
+    /// <summary>
+    /// Represents a comparable and equatable reference.
+    /// </summary>
+    /// <typeparam name="T">The type of the value.</typeparam>
+    public abstract class OpenXmlComparableReferenceType<T> : OpenXmlSimpleType,
+        IComparable, IComparable<OpenXmlComparableReferenceType<T>>, IEquatable<OpenXmlComparableReferenceType<T>>
+        where T : class, IComparable, IComparable<T>, IEquatable<T>
+    {
+        /// <summary>
+        /// Creates a new instance of <see cref="OpenXmlComparableReferenceType{T}"/>.
+        /// </summary>
+        private protected OpenXmlComparableReferenceType()
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="OpenXmlComparableReferenceType{T}"/>
+        /// class by deep copying the supplied <see cref="OpenXmlComparableReferenceType{T}"/>
+        /// value.
+        /// </summary>
+        /// <param name="source">The source <see cref="OpenXmlComparableReferenceType{T}"/> instance.</param>
+        private protected OpenXmlComparableReferenceType(OpenXmlComparableReferenceType<T> source) : base(source)
+        {
+        }
+
+        /// <summary>
+        /// Gets or sets the value of this instance.
+        /// </summary>
+        public abstract T Value { get; set; }
+
+        /// <inheritdoc />
+        public int CompareTo(object obj)
+        {
+            if (obj is OpenXmlComparableReferenceType<T> other)
+            {
+                return CompareTo(other);
+            }
+
+            throw new ArgumentException();
+        }
+
+        /// <inheritdoc />
+        public int CompareTo(OpenXmlComparableReferenceType<T> other)
+        {
+            if (other is null)
+            {
+                return 1;
+            }
+
+            if (Value == null)
+            {
+                return other.Value == null ? 0 : -1;
+            }
+
+            return Value.CompareTo(other.Value);
+        }
+
+        /// <inheritdoc />
+        public bool Equals(OpenXmlComparableReferenceType<T> other)
+        {
+            if (other is null)
+            {
+                return false;
+            }
+
+            if (Value == null)
+            {
+                return other.Value == null;
+            }
+
+            return Value.Equals(other.Value);
```

Comment:
Can this be collapsed into `Equals(Value, other?.Value)`?

Reply:
That depends on whether a non-null instance with a null Value property should be equal to a null instance.

For example, I could have two non-null `StringValue` instances, with `Value` being null for both of them. Those two should be considered equal. However, when I take any of those, I'd say that `stringValue.Equals(null)` should be false. In this case, the shortened `Equals` method could look like this:
```
        public bool Equals(OpenXmlComparableReferenceType<T> other)
        {
            return !(other is null) && Equals(Value, other.Value);
        }
```

Reply:
Hmm I still think that `Equals(Value, other?.Value)` would work, but I guess I'm not sure off the top of my head what `Equals(null,null)` is. Do you have unit tests covering the cases you bring up? If so you can just try it out. I'll leave this up to you - no need to rathole into it as the current implementation works

Reply:
`Equals(null, null)` should be true.